infra/resources/modules/commons/appgateway.tf

```diff
@@ -14,4 +14,292 @@ resource "azurerm_public_ip" "appgateway_public_ip" {
   zones               = [1, 2, 3]
 
   tags = var.tags
+}
+
+# Subnet to host the application gateway
+module "appgateway_snet" {
+  source                                    = "git::https://github.com/pagopa/terraform-azurerm-v3.git//subnet?ref=v8.20.0"
+  name                                      = format("%s-appgateway-snet", local.project)
+  address_prefixes                          = var.cidr_subnet_appgateway
+  resource_group_name                       = azurerm_resource_group.rg_external.name
+  virtual_network_name                      = module.vnet.name
+  private_endpoint_network_policies_enabled = true
+
+  service_endpoints = [
+    "Microsoft.Web",
+  ]
+}
+
+## Application gateway ##
+module "app_gw" {
+  source = "github.com/pagopa/terraform-azurerm-v3.git//app_gateway?ref=v8.20.0"
+
+  resource_group_name = azurerm_resource_group.rg_external.name
+  location            = azurerm_resource_group.rg_external.location
+  name                = format("%s-appgateway", local.project)
+  zones               = [1, 2, 3]
+
+  # SKU
+  sku_name = "WAF_v2"
+  sku_tier = "WAF_v2"
+
+  # Networking
+  subnet_id    = module.appgateway_snet.id
+  public_ip_id = azurerm_public_ip.appgateway_public_ip.id
+
+  # Configure backends
+  backends = {
+
+    apim = {
+      protocol                    = "Https"
+      host                        = "google.com"
+      port                        = 443
+      ip_addresses                = null # with null value use fqdns
+      fqdns                       = ["google.com"]
```

Comment:
**question**: Are `host` and `fqdns` correct?

Reply:
nope, it's a kind of placeholder, we must replace them with APIM's related host and fqdn once we merge APIM PR :)

Reply:
The APIM PR is ready to be reviewer and merged :)